src/main/java/graphql/schema/idl/SchemaParseOrder.java

```diff
@@ -0,0 +1,149 @@
+        });
+        return named;
+    }
+
+    /**
+     * This comparator will sort according to the original parsed order
+     *
+     * @param <T> is a {@link GraphQLSchemaElement}
+     *
+     * @return a comparator that sorts schema elements in parsed order
+     */
+    public <T extends GraphQLSchemaElement> Comparator<? super T> getElementComparator() {
+        // relies in the fact that names in graphql schema are unique across ALL elements
+        Map<String, Integer> namedSortValues = buildNameIndex(getInNameOrder());
+        return (e1, e2) -> {
+            if (e1 instanceof GraphQLFieldDefinition || e1 instanceof GraphQLInputObjectField || e1 instanceof GraphQLEnumValueDefinition) {
+                return 0; // as is
+            }
+            if (e1 instanceof GraphQLNamedSchemaElement && e2 instanceof GraphQLNamedSchemaElement) {
+                int sortVal1 = sortVale((GraphQLNamedSchemaElement) e1, namedSortValues);
+                int sortVal2 = sortVale((GraphQLNamedSchemaElement) e2, namedSortValues);
+                return Integer.compare(sortVal1, sortVal2);
+            }
+            return -1; // sort to the top
+        };
+    }
+
+    private Integer sortVale(GraphQLNamedSchemaElement e1, Map<String, Integer> namedSortValues) {
+        return namedSortValues.getOrDefault(e1.getName(), -1);
+    }
+
+    private Map<String, Integer> buildNameIndex(Map<String, List<SDLNamedDefinition<?>>> inNameOrder) {
+        Map<String, Integer> nameIndex = new HashMap<>();
+        int sourceIndex = 0;
+        for (Map.Entry<String, List<SDLNamedDefinition<?>>> entry : inNameOrder.entrySet()) {
+            List<SDLNamedDefinition<?>> namedDefs = entry.getValue();
+            for (int i = 0; i < namedDefs.size(); i++) {
+                SDLNamedDefinition<?> namedDefinition = namedDefs.get(i);
+                // we will put it in parsed order AND with first sources first
+                int index = i + (sourceIndex * 100_000_000);
+                nameIndex.put(namedDefinition.getName(), index);
+            }
+            sourceIndex++;
+        }
+        return nameIndex;
+    }
+
+    /**
+     * This adds a new {@link SDLDefinition} to the order
+     *
+     * @param sdlDefinition the SDL definition to add
+     * @param <T>           for two
+     *
+     * @return this {@link SchemaParseOrder} for fluent building
+     */
+    public <T extends SDLDefinition<?>> SchemaParseOrder addDefinition(T sdlDefinition) {
+        if (sdlDefinition != null) {
+            definitionList(sdlDefinition).add(sdlDefinition);
+        }
+        return this;
+    }
+
+    /**
+     * This removes a {@link SDLDefinition} from the order
+     *
+     * @param sdlDefinition the SDL definition to remove
+     * @param <T>           for two
+     *
+     * @return this {@link SchemaParseOrder} for fluent building
+     */
+    public <T extends SDLDefinition<?>> SchemaParseOrder removedDefinition(T sdlDefinition) {
```

Comment:
typo (I think): should be `removeDefinition`

Reply:
done